src/main/osd/osd_elements.c

```diff
@@ -1413,8 +1412,20 @@ static bool osdDrawSingleElement(displayPort_t *osdDisplayPort, uint8_t item)
 
 void osdDrawActiveElements(displayPort_t *osdDisplayPort, timeUs_t currentTimeUs)
 {
+#ifdef USE_GPS
+    static bool lastGpsSensorState;
+    // Handle the case that the GPS_SENSOR may be delayed in activation
+    // or deactivate if communication is lost with the module.
+    const bool currentGpsSensorState = sensors(SENSOR_GPS);
+    if (lastGpsSensorState != currentGpsSensorState) {
+        lastGpsSensorState = currentGpsSensorState;
+        osdAnalyzeActiveElements();
+    }
+#endif // USE_GPS
+
     blinkState = (currentTimeUs / 200000) % 2;
 
+
```

Comment:
Extra newline.

Reply:
Argh! Originally had the added logic below `blinkState` and I rearranged it. Copy/pasting too early in the morning. I'll fix.